scilpy/tracking/propagator.py

```diff
@@ -260,16 +438,82 @@ def _sample_next_direction(self, pos, v_in):
         Return
         ------
         direction: ndarray (3,)
-            The maxima closest to v_in. None if the no
-            valid maxima are available.
-        """
-        possible_maxima = self.tracking_field.get_possible_next_directions(
-            pos, v_in, 'det')
-        cosinus = 0
-        v_out = None
-        for d in possible_maxima:
-            new_cosinus = np.dot(v_in, d)
-            if new_cosinus > cosinus:
-                cosinus = new_cosinus
-                v_out = d
-        return v_out
+            A valid tracking direction. None if no valid direction is found.
+        """
+        if self.algo == 'prob':
+            # Tracking field returns the sf and directions
+            sf, directions = self._get_possible_next_dirs_prob(pos, v_in)
+
+            # Sampling one.
+            if np.sum(sf) > 0:
+                v_out = directions[sample_distribution(sf)]
+            else:
+                return None
+        elif self.algo == 'det':
+            # Tracking field returns the list of possible maxima.
+            possible_maxima = self._get_possible_next_dirs_det(pos, v_in)
+            # Choosing one.
+            cosinus = 0
+            v_out = None
+            for d in possible_maxima:
+                new_cosinus = np.dot(v_in, d)
+                if new_cosinus > cosinus:
+                    cosinus = new_cosinus
+                    v_out = d
+        else:
+            raise ValueError("Tracking choice must be one of 'det' or 'prob'.")
+
+        # Normalizing result? Coming from dipy's sphere so supposing that ok.
```

Comment:
Yes let's assume `self.sphere` is always the unit sphere. We could normalize in constructor to make sure, but that's not necessary, I think.